src/components/Modal/Modal.tsx

```diff
@@ -88,20 +106,27 @@ export function Modal({
                 appear={true}
                 onEnter={() => setInTransition(true)}
                 onExit={() => setInTransition(true)}
-                onEntered={() => setInTransition(false)}
-                onExited={() => setInTransition(false)}
+                onEntered={() => {
+                    setInTransition(false);
+                    onOpenEnd?.();
+                }}
+                onExited={() => {
+                    setInTransition(false);
+                    onCloseEnd?.();
```

Comment:
Maybe add lifecycle callbacks for all events? I would name it like so: `onTransitionEnter`, `onTransitionEntered`, `onTransitionExit`, `onTransitionExited`